src/main/java/com/kumuluz/ee/config/microprofile/adapters/ConfigSourceAdapter.java

```diff
@@ -55,10 +58,17 @@ public ConfigSourceAdapter(ConfigurationSource configurationSource) {
     public int getOrdinal() {
         return configurationSource.getOrdinal();
     }
-
+    
     @Override
     public String getValue(String s) {
-        return configurationSource.get(s).orElse(null);
+    	String val = configurationSource.get(s).orElse(null);
```

Comment:
We cannot rely on the curly braces format here, that is specific of yaml configuration source. Env and system property configuration sources return Optional.empty() if get() for list is called.

Instead the way to go here is to check if the value is a list via getListSize and if it is iterating through values (while validating there are no nested objects) and collecting them to a comma separated string.

Reply:
>>We cannot rely on the curly braces format here, that is specific of yaml configuration source

Not sure i get what you mean: there're no curly braces anywhere and "[, ]" notion is the Java common string representation for arrays.

>>iterating through values

But it's a string, how can i iterate through values without parsing it first?

Reply:
>>iterating through values (while validating there are no nested objects) and collecting them to a comma separated string.

In fact there's no need to do any of that: the corresponding ConfigSource has done it already. The value you get from it is the leaf value of YAML tree which is either a primitive, a String or a collection of the above.

Reply:
Yes, but the value isn't necessary from a yaml source, it can be an environment variable, system property, etcd value or even something implemented by the user.

See here for an example: https://github.com/kumuluz/kumuluzee/blob/392d71b11dae349bf002f3c2b854943c3db5e062/core/src/main/java/com/kumuluz/ee/factories/EeConfigFactory.java#L122

Reply:
Ah, you mean that supplying arrays is only supported by `FileConfigurationSource` and not by other two that kumuluzee-common has (and the etcd one)? Indeed it's true, but imho `ConfigSourceAdapter` isn't the right place to implement that precisely because defining an array/list may well be source-specific and you will never get it right in a common adapter.

I think we must agree upon the contract that a `com.kumuluz.ee.configuration.ConfigurationSource` must deal with it within itself (i will modify the method JavaDoc to mention that) and provide an array string representation from #get(String) (as the file one does). Then we implement that in each source (which is architecturally sound and #getListSize(String) in each of the sources already does all of the array parsing work anyway) and my fix does just what it does now.

What do you think?

Reply:
Defining array isn't source-specific, it is well defined: https://github.com/kumuluz/kumuluzee/wiki/Configuration#configuration-parameter-keys

It has to be because the whole configuration framework is organized in a hierarchy and if you want to override a key in a higher priority source the keys have to be same.

Reply:
Ah, now i see what you mean. Then we will have to accept that for YAML sources we are doing same work twice, because the array that `FileConfigurationSource` has read from config.yaml has already been analysed, collected and processed.

I will do as you suggested, though I still think that the code snippet you refered to: https://github.com/kumuluz/kumuluzee/blob/392d71b11dae349bf002f3c2b854943c3db5e062/core/src/main/java/com/kumuluz/ee/factories/EeConfigFactory.java#L122 is not the right way this must be done; you must be able to get an array of values from a ConfigSource because the low-level contracts for arrays/lists are different (e.g. storing in the database or Zookeeper trees, or even YAML) and the end-user should not do all that stuff iterating through values.